sormas-backend/src/main/java/de/symeda/sormas/backend/document/DocumentFacadeEjb.java

```diff
@@ -0,0 +1,154 @@
+
+	@PersistenceContext(unitName = ModelConstants.PERSISTENCE_UNIT_NAME)
+	private EntityManager em;
+
+	@EJB
+	private UserService userService;
+	@EJB
+	private DocumentService documentService;
+	@EJB
+	private DocumentStorageService documentStorageService;
+
+	@Inject
+	private Event<DocumentSaved> documentSavedEvent;
+
+	@Override
+	public DocumentDto getDocumentByUuid(String uuid) {
+		return toDto(documentService.getByUuid(uuid));
+	}
+
+	@Override
+	public DocumentDto saveDocument(DocumentDto dto, byte[] content) throws IOException {
+		Document existingDocument = dto.getUuid() == null ? null : documentService.getByUuid(dto.getUuid());
+		if (existingDocument != null) {
+			// TODO: add exception message
+			throw new EntityExistsException();
+		}
+
+		Document document = fromDto(dto);
+
+		document.setStorageReference(documentStorageService.save(document, content));
+
+		documentService.persist(document);
+		documentService.doFlush();
+
+		documentSavedEvent.fire(new DocumentSaved(document));
+
+		return toDto(document);
+	}
+
+	@Override
+	public void deleteDocument(String uuid) {
+		// Only mark as delete here; actual deletion will be done in document storage cleanup via cron job
+		documentService.markAsDeleted(documentService.getByUuid(uuid));
+	}
+
+	@Override
+	public List<DocumentDto> getDocumentsRelatedToEntity(DocumentRelatedEntityType type, String uuid) {
+		return documentService.getRelatedToEntity(type, uuid).stream().map(DocumentFacadeEjb::toDto).collect(Collectors.toList());
+	}
+
+	@Override
+	public String isExistingDocument(DocumentRelatedEntityType type, String uuid, String name) {
+		return documentService.isExisting(type, uuid, name);
+	}
+
+	@Override
+	public byte[] read(String uuid) throws IOException {
+		Document document = documentService.getByUuid(uuid);
+		return documentStorageService.read(document.getStorageReference());
+	}
+
+	@Override
+	public void cleanupDeletedDocuments() {
+		List<Document> deleted = documentService.getDeletedDocuments();
+		for (Document document : deleted) {
+			documentStorageService.delete(document.getStorageReference());
+			documentService.delete(document);
+		}
+	}
+
+	public Document fromDto(DocumentDto source) {
+		Document target = documentService.getByUuid(source.getUuid());
+		if (target == null) {
+			target = new Document();
+			target.setUuid(source.getUuid());
+			if (source.getCreationDate() != null) {
+				target.setCreationDate(new Timestamp(source.getCreationDate().getTime()));
+			}
+		}
+		DtoHelper.validateDto(source, target);
+
+		target.setUploadingUser(userService.getByReferenceDto(source.getUploadingUser()));
+		target.setName(source.getName());
+		target.setContentType(source.getContentType());
+		target.setSize(source.getSize());
+		target.setRelatedEntityUuid(source.getRelatedEntityUuid());
+		target.setRelatedEntityType(source.getRelatedEntityType());
+
+		return target;
+	}
+
+	// XXX: pseudonimize uploadingUser? Under which conditions?
```

Comment:
@tbroyer The pseudonymization depends on the related entity type. I'd suggest to create a convertToDto and pseudonymizeDto method as we have in other facades.
In the pseudonymizeDto  have a switch on the related entity type. For events you will basically do the same as in EventFacadeEjb.pseudonymizeDto

Reply:
Done